src/nvofapi_cuda.cpp

```diff
@@ -0,0 +1,58 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: MIT
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+#include "util/util_env.h"
+#include "util/util_log.h"
+#include "util/util_string.h"
+#include "util/util_statuscode.h"
+#include "../version.h"
+#include "../config.h"
+
+extern "C" {
+
+    using namespace dxvk;
+
+    // CUDA entrypoint. Returning error with a message to upgrade DLSS version
+    NV_OF_STATUS __cdecl NvOFAPICreateInstanceCuda(uint32_t apiVer, void* functionList) {
```

Comment:
I suggest to do this without all the ceremony:

```c++
    NV_OF_STATUS NVOFAPI NvOFAPICreateInstanceCuda(uint32_t apiVer, void* functionList) {
        constexpr auto n = __func__;

        if (log::tracing())
            log::trace(n, apiVer, log::fmt::ptr(functionList));

        return UnsupportedFeature(n);
    }
```

Reply:
I would prefer a "not supported" tho.. and would not hurt to create one for NvOFAPICreateInstanceD3D11 too.. just for completeness.. Instead of making suggestions to replace DLSS i mean.
Gives the USER a bit more understanding? :)

`log::info(str::format("OpticalFlow CUDA is not supported."));`

instead of just silently dropping?

Reply:
Well.. depends on the output of "UnsupportedFeature" i guess.. have not looked at the actual returntext when that happens? hmm...

Reply:
Returning `NV_OF_ERR_UNSUPPORTED_FEATURE` says exactly that we do not support this feature, this should be sufficient imho, no need to add more logging to that. Agree on doing the same for D3D11.

Reply:
Looking at the error code description, `NV_OF_ERR_OF_NOT_AVAILABLE` is even better:

```c++
    /**
    * This indicates that HW Optical flow functionality is not supported
    */
    NV_OF_ERR_OF_NOT_AVAILABLE,
```

Reply:
Ah, seeing the `void*` just now… that’s not a good idea imho, we should stick to the headers, but those have a cuda dependency, that’s not gonna work without some ugly things I prefer to avoid.